wagtail/snippets/views/snippets.py

```diff
@@ -74,96 +75,107 @@ def index(request):
         raise PermissionDenied
 
 
-def list(request, app_label, model_name):
-    model = get_snippet_model_from_url_params(app_label, model_name)
+class BaseListView(TemplateView):
```

Comment:
If we're going to go with generics here, is there any reason we don't go with `ListView` for this, or perhaps even our own [`wagtail.admin.views.generic.model.IndexView`](`https://github.com/wagtail/wagtail/blob/main/wagtail/admin/views/generic/models.py#L14`)?

It feels like a missed opportunity to take this step without breaking the code down into more digestible (and ultimately more overridable) chunks.

Reply:
No, the missed opportunity would have been if I'd sat on this PR indefinitely because I didn't have several days free to do all the refactoring that could be done here :-) Incremental changes are the way to go (and more friendly to code reviewers).

There is indeed a lot more that could be done here, and I hope that in time this will eventually become a generic listing view. I think the best way to get to that point without breaking anything is to start from the lowest level (initially just moving the existing code into a `dispatch` method, not even splitting it into GET/POST cases) and progressing to higher levels from there.

Reply:
Okay, happy with that approach.